### PR TITLE
defering deletion of children in EditorInspector::_clear.

### LINKED ISSUE #20501: Godot randomly crash when I click in "open resource"
**Godot version:**
936b5a395a4bd80fbb2832ff175efbde844e5fde

**OS/device including version:**
Linux 4.17.9
GeForce GTX 970

**Issue description:**
Godot randomly crash when I click in "open resource" to edit `Shader` or `VisualShader`.

```
OpenGL ES 3.0 Renderer: GeForce GTX 970/PCIe/SSE2
OPENING: /home/guilherme/Workspaces/Loop Test (::home::guilherme::Workspaces::Loop Test)
OpenGL ES 3.0 Renderer: GeForce GTX 970/PCIe/SSE2
┌[20:32:24] guilherme:~
└$ ERROR: generate: EditorResourcePreviewGenerator::generate needs to be overridden
   At: editor/editor_resource_preview.cpp:55.
ERROR: generate: EditorResourcePreviewGenerator::generate needs to be overridden
   At: editor/editor_resource_preview.cpp:55.
ERROR: generate: EditorResourcePreviewGenerator::generate needs to be overridden
   At: editor/editor_resource_preview.cpp:55.
ERROR: generate: EditorResourcePreviewGenerator::generate needs to be overridden
   At: editor/editor_resource_preview.cpp:55.
handle_crash: Program crashed with signal 11
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[1] /usr/lib/libc.so.6(+0x368f0) [0x7f92dde078f0] (??:0)
[2] Set<BaseButton*, Comparator<BaseButton*>, DefaultAllocator>::front() const (??:0)
[3] BaseButton::_unpress_group() (??:0)
[4] BaseButton::_gui_input(Ref<InputEvent>) (??:0)
[5] MethodBind1<Ref<InputEvent> >::call(Object*, Variant const**, int, Variant::CallError&) (??:0)
[6] Object::call_multilevel(StringName const&, Variant const**, int) (??:0)
[7] Object::call_multilevel(StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (??:0)
[8] Viewport::_gui_call_input(Control*, Ref<InputEvent> const&) (??:0)
[9] Viewport::_gui_input_event(Ref<InputEvent>) (??:0)
[10] Viewport::input(Ref<InputEvent> const&) (??:0)
[11] Viewport::_vp_input(Ref<InputEvent> const&) (??:0)
[12] MethodBind1<Ref<InputEvent> const&>::call(Object*, Variant const**, int, Variant::CallError&) (??:0)
[13] Object::call(StringName const&, Variant const**, int, Variant::CallError&) (??:0)
[14] Object::call(StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (??:0)
[15] SceneTree::call_group_flags(unsigned int, StringName const&, StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (??:0)
[16] SceneTree::input_event(Ref<InputEvent> const&) (??:0)
[17] InputDefault::_parse_input_event_impl(Ref<InputEvent> const&, bool) (??:0)
[18] InputDefault::parse_input_event(Ref<InputEvent> const&) (??:0)
[19] OS_X11::process_xevents() (??:0)
[20] OS_X11::run() (??:0)
[21] /home/guilherme/Workspaces/godot/bin/godot.x11.tools.64(main+0xd5) [0x55eae590b55a] (??:0)
[22] /usr/lib/libc.so.6(__libc_start_main+0xeb) [0x7f92dddf406b] (??:0)
[23] /home/guilherme/Workspaces/godot/bin/godot.x11.tools.64(_start+0x2a) [0x55eae590b3ca] (??:0)
-- END OF BACKTRACE --

```

**Steps to reproduce:**
Create a Shader or VisualShader;
Click in "Open Reource" some times until crash.

**Minimal reproduction project:**
[Loop Test.zip](https://github.com/godotengine/godot/files/2233913/Loop.Test.zip)

### PR DESCRIPTION
**Pull request by ibrahn**
Nodes were being deleted with calls pending.

fixes #20501

### CODE REVIEW

**Comment by AlexHolly** [comment]
It seems to fix a case where the "Open Editor" Button caused a crash when the target icon+.import file was manually deleted on the filesystem.

File: editor/editor_inspector.cpp [hunk 1]
```diff
@@ -1701,9 +1701,12 @@ void EditorInspector::update_property(const String &p_prop) {
 
 void EditorInspector::_clear() {
 
-	while (main_vbox->get_child_count()) {
-		memdelete(main_vbox->get_child(0));
+	for (int i = main_vbox->get_child_count(); i-- > 0;) {
```
Comment (akien-mga) [inline editor/editor_inspector.cpp]: Please use `i > 0; i--` instead of this shortcut, it's what we use everywhere else.

Comment (ibrahn) [reply editor/editor_inspector.cpp]: Updated to `int i = main_vbox->get_child_count() - 1; i >= 0; i--` to match style

(For reference, the previous version isn't a shortcut. It's the form of the loop that also works for an unsigned type like size_t)

Comment (akien-mga) [reply editor/editor_inspector.cpp]: Ah thanks, that makes sense and it's good to know :)

**Comment by akien-mga** [comment]
Superseded by #21015, the bug was a regression from #20501.

**Pull request closed without merging**